eth-connector/src/lib.rs

```diff
@@ -247,46 +261,19 @@ impl EngineFungibleToken for EthConnectorContract {
         msg: String,
     ) -> PromiseOrValue<U128> {
         self.assert_access_right().sdk_unwrap();
-        self.register_if_not_exists(&receiver_id);
-
-        let amount: Balance = amount.into();
-        crate::log!(
-            "Transfer call from {} to {} amount {}",
-            sender_id,
-            receiver_id,
-            amount,
-        );
-
-        // Verify message data before `ft_on_transfer` call to avoid verification panics
-        // It's allowed empty message if `receiver_id =! current_account_id`
-        if sender_id == receiver_id {
-            let message_data = FtTransferMessageData::parse_on_transfer_message(&msg).sdk_unwrap();
-            // Check is transfer amount > fee
-            if message_data.fee.as_u128() >= amount {
-                panic_err("insufficient balance for fee");
-            }
-        }
+        self.internal_ft_transfer_call(sender_id, receiver_id, amount, memo, msg)
+    }
+}
 
-        // Special case for Aurora transfer itself - we shouldn't transfer
-        if sender_id != receiver_id {
-            self.ft
-                .internal_transfer(&sender_id, &receiver_id, amount, memo);
-        }
+/// Management for a known Engine accounts
+#[near_bindgen]
+impl KnownEnginAccountsManagement for EthConnectorContract {
+    fn set_engine_account(&mut self, engine_account: AccountId) {
```

Comment:
This function should probably be access controlled in some way, otherwise a malicious actor could add a non-engine as a known engine and thus prevent `ft_transfer_call` from working with that account because users would not expect to need to set the message field for a non-engine.